BaragonAgentService/src/main/java/com/hubspot/baragon/agent/healthcheck/InternalStateChecker.java

```diff
@@ -46,7 +69,7 @@ public InternalStateChecker(BaragonStateDatastore stateDatastore,
   public void run() {
     Set<String> invalidServiceMessages = new HashSet<>();
     long now = System.currentTimeMillis();
-    internalStateCache.forEach((serviceId, context) -> {
+    new HashMap<>(internalStateCache).forEach((serviceId, context) -> {
```

Comment:
Any potential issues newing this each run? (How big does this cache get?)

Reply:
Good Q, need to check on size for some of the larger groups. Wanted to be able to update the initial source without waiting for the full run through the list to be done which is why I made a copy